src/utils/hooks/useLastQueryExecutionSettings.ts

```diff
@@ -9,13 +10,7 @@ export const useLastQueryExecutionSettings = () => {
     );
 
     const lastSettings: QuerySettings | undefined = lastStorageSettings
-        ? {
-              transactionMode: lastStorageSettings.transactionMode,
-              queryMode: lastStorageSettings.queryMode,
-              statisticsMode: lastStorageSettings.statisticsMode,
-              tracingLevel: lastStorageSettings.tracingLevel,
-              timeout: lastStorageSettings.timeout,
-          }
+        ? querySettingsValidationSchema.parse(lastStorageSettings)
```

Comment:
This may throw.
```ts
let lastSettings: QuerySettings | undefined;
try {
    lastSettings = querySettingsValidationSchema.parse(lastStorageSettings);
catch () {
    lastSettings = undefined;
}